tests/functionaltests/Tests/UIKitTests/UIActivityIndicatorTests.mm

```diff
@@ -63,4 +72,479 @@
+            EXPECT_EQ(50, xamlElement.Width());
+
+            // The ProgressRing should always be centered in the element
+            ASSERT_TRUE(isCenteredInParent(progressRing, xamlElement));
+        });
+    }
+
+    TEST_METHOD(UIActivityIndicatorView_StartAnimating) {
+        StrongId<UIActivityIndicatorViewController> activityIndicatorVC;
+        activityIndicatorVC.attach([[UIActivityIndicatorViewController alloc] init]);
+
+        UXTestAPI::ViewControllerPresenter testHelper(activityIndicatorVC, 2);
+        UIActivityIndicatorView* indicator = [activityIndicatorVC indicator];
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            FrameworkElement xamlElement = [indicator _winrtXamlElement];
+            Microsoft::WRL::ComPtr<IInspectable> inspectable(XamlGetInternalProgressRing(objcwinrt::to_insp(xamlElement)));
+            auto progressRing = objcwinrt::from_insp<Controls::ProgressRing>(inspectable);
+            ASSERT_TRUE(progressRing);
+
+            [indicator startAnimating];
+            ASSERT_TRUE(indicator.isAnimating);
+
+            ASSERT_TRUE(progressRing.IsActive());
+            EXPECT_EQ(winrt::Windows::UI::Xaml::Visibility::Visible, progressRing.Visibility());
+            ASSERT_FALSE(indicator.hidden);
+            EXPECT_EQ(1.0, xamlElement.Opacity());
+            EXPECT_EQ(1.0, indicator.alpha);
+        });
+    }
+
+    TEST_METHOD(UIActivityIndicatorView_StopAnimating) {
+        StrongId<UIActivityIndicatorViewController> activityIndicatorVC;
+        activityIndicatorVC.attach([[UIActivityIndicatorViewController alloc] init]);
+
+        UXTestAPI::ViewControllerPresenter testHelper(activityIndicatorVC, 2);
+        UIActivityIndicatorView* indicator = [activityIndicatorVC indicator];
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            FrameworkElement xamlElement = [indicator _winrtXamlElement];
+            Microsoft::WRL::ComPtr<IInspectable> inspectable(XamlGetInternalProgressRing(objcwinrt::to_insp(xamlElement)));
+            auto progressRing = objcwinrt::from_insp<Controls::ProgressRing>(inspectable);
+
+            [indicator startAnimating];
+            ASSERT_TRUE(indicator.isAnimating);
+            ASSERT_TRUE(progressRing.IsActive());
+            EXPECT_EQ(winrt::Windows::UI::Xaml::Visibility::Visible, xamlElement.Visibility());
+
+            [indicator stopAnimating];
+            ASSERT_FALSE(indicator.isAnimating);
+            ASSERT_FALSE(progressRing.IsActive());
+            // Element should be still "Visible" after stopping animation, to ensure space is preserved in the layout.
+            EXPECT_EQ(winrt::Windows::UI::Xaml::Visibility::Visible, xamlElement.Visibility());
+            ASSERT_FALSE(indicator.hidden);
+            EXPECT_EQ(0.0, xamlElement.Opacity());
+            EXPECT_EQ(0.0, indicator.alpha);
+        });
+    }
+
+    TEST_METHOD(UIActivityIndicatorView_HidesWhenStopped_BeforeStartAnimation) {
```

Comment:
>UIActivityIndicatorView_HidesWhenStopped_BeforeStartAnimation [](start = 16, length = 61)

Some comments about expected behavior on this test and others (where not obvious) about expected behavior would be good.